### ISSUE TITLE
Errors when updating to .04.0-v

### ISSUE THREAD

**Issue by PathfinderNetworks** [Ylianst/MeshCentral#487]
I'm trying to update my MeshCentral server from 0.4.0-k to 0.4.0-v.
I'm updating via command prompt on a Server 2019 box using npm install meshcentral within the MeshCentral directory (like I've done for past versions).  
This upgrade completes and then I restart the MeshCentral service.
However, it never finishes loading.   What I'm seeing, when watching Task Manager, is the Node.js: Server-side Javascript tasks fire up.  Three of them.  But what happens is that one of those three tasks will quickly ramp up to 100% cpu usage (which is fairly normal when first starting Mesh).  Then that one task will terminate. Then, a couple seconds later, that same task starts back up again and quickly goes back up to 100% cpu usage. Then it terminates.  This loop continues non-stop. If I terminate the MeshCentral service then all Node.js tasks quit (as expected). 
If I roll back to 0.4.0-k it works fine. 
I'm seeing this in the mesherrors.tx file.  Looks like something to do with the SMTP settings?  Those settings work correctly under 0.4.0-k.  

-------- 9/15/2019, 7:09:54 PM ---- 0.4.0-v --------

c:\Program Files\Open Source\MeshCentral\node_modules\meshcentral\meshmail.js:221
                console.log('SMTP mail server ' + parent.config.smtp.host + ' failed: ' + JSON.stringify(err));
                                                                                               ^

TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at c:\Program Files\Open Source\MeshCentral\node_modules\meshcentral\meshmail.js:221:96
    at SMTPConnection.connection.once.err (c:\Program Files\Open Source\MeshCentral\node_modules\nodemailer\lib\smtp-transport\index.js:346:24)
    at Object.onceWrapper (events.js:285:13)
    at SMTPConnection.emit (events.js:197:13)
    at SMTPConnection._onError (c:\Program Files\Open Source\MeshCentral\node_modules\nodemailer\lib\smtp-connection\index.js:771:14)
    at TLSSocket.SMTPConnection._onSocketError (c:\Program Files\Open Source\MeshCentral\node_modules\nodemailer\lib\smtp-connection\index.js:190:47)
    at TLSSocket.emit (events.js:197:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at processTicksAndRejections (internal/process/next_tick.js:76:17)

**Comment by PathfinderNetworks** [comment]
Also, I run my own email server so I can see the detailed logs.  I can see that MeshCentral is connecting to the email server but then closes the connection for some reason.
Sun 2019-09-15 19:09:43.665: 05: Accepting SMTP connection from 10.100.xx.x:51929 to 10.100.xx.y:465
Sun 2019-09-15 19:09:43.710: 01: SSL negotiation successful (TLS 1.2, 384 bit key exchange, 256 bit AES encryption)
Sun 2019-09-15 19:09:43.710: 03: --> 220 myemaildomain.com ESMTP Sun, 15 Sep 2019 19:09:43 -0400
Sun 2019-09-15 19:09:43.826: 05: Connection closed
Sun 2019-09-15 19:09:43.826: 04: SMTP session terminated (Bytes in/out: 408/3577)

I've obfuscated the IP address and email domain in the above logs from my email server.

**Comment by PathfinderNetworks** [comment]
Ok, I figured this out.  There must be more advanced TLS certificate checking on the SMTP side of things with this new version.  I looked at my config.json and could see that I was using the IP local address of the email server instead of the domain name. The SSL certificate would have then shown as not being valid for that SMTP server because it wouldn't match the IP address (since it's a LAN IP address and I use a signed certificate for my SMTP server).  I changed the SMTP host entry in config.json to the actual domain name of the server (so the certificate will match) and it's now working.

**Comment by Ylianst** [comment]
I got one meeting to attend for the next hour, after that, I am working on this SMTP problem right away. I expect to have a fix in a few hours. Basically, I will add an option to accept the SMTP server TLS cert even if not "valid".

**Comment by Ylianst** [comment]
Just published MeshCentral v0.4.0-w with two extra options on the "smtp" configuration in the config.json file.

```
  "smtp": {
    "host": "smtp.server.com",
    "port": 25,
    "from": "mymail@server.com",
    "tls": false,
    "tlscertcheck": false,
    "tlsstrict": false
  }
```

The two new options are "tlscertcheck" (default true) and "tlsstrict" (default false). If the SMTP server uses a self-signed or untrusted TLS certificate, set "tlscertcheck" to false to have MeshCentral connect to it anyway. Setting "tlsstrict" to true just disallows older SSLv2/SSLv3, etc.

Also note that setting "tls" to true will initiate TLS right away, but often SMTP servers start the connection without TLS and do a STARTTLS to switch to TLS. So setting "tls" to false is often correct and still secure.

I suggest trying with the following to start:

```
    "tls": false,
    "tlscertcheck": false,
    "tlsstrict": false
```

Let me know what happens.

**Comment by Ylianst** [comment]
Closing this one since it seems to be fixed. Please re-open if needed.
